Index: trousers/src/tcsd/platform.c
===================================================================
--- trousers.orig/src/tcsd/platform.c
+++ trousers/src/tcsd/platform.c
@@ -13,7 +13,7 @@
 #include <sys/param.h>
 #include <sys/sysctl.h>
 #include <err.h>
-#elif (defined (__linux) || defined (linux) || defined(__GLIBC__))
+#elif (defined (__linux) || defined (linux) || defined(__GNU__) || defined(__GLIBC__))
 #include <utmp.h>
 #elif (defined (SOLARIS))
 #include <utmpx.h>
